ddl/index.go

```diff
@@ -161,35 +162,45 @@ func checkIndexColumn(col *model.ColumnInfo, ic *ast.IndexColName) error {
 }
 
 func getIndexColumnLength(col *model.ColumnInfo, colLen int) (int, error) {
-	// Take care of the sum of length of all index columns.
-	if colLen != types.UnspecifiedLength {
-		return colLen, nil
-	}
-	// Specified data types.
-	if col.Flen != types.UnspecifiedLength {
-		// Special case for the bit type.
-		if col.FieldType.Tp == mysql.TypeBit {
-			return (col.Flen + 7) >> 3, nil
+	length := takeIfSpecifiedLength(colLen, col.Flen)
+
+	switch col.Tp {
+	case mysql.TypeBit:
+		return (length + 7) >> 3, nil
+	case mysql.TypeVarchar, mysql.TypeString:
+		for _, d := range charset.GetSupportedCharsets() {
```

Comment:
I think it's better to use `charset.GetCharsetDesc()` here, this function has been used for column length validation somewhere.

Reply:
Done. PTAL